src/Microsoft.Performance.Toolkit.Engine/EngineCreateInfo.cs

```diff
@@ -38,6 +39,52 @@ public EngineCreateInfo(ReadOnlyDataSourceSet dataSources)
             Guard.NotNull(dataSources, nameof(dataSources));
 
             this.DataSources = dataSources;
+            this.OptionsResolver = new GlobalProcessingOptionsResolver(ProcessorOptions.Default);
+        }
+
+        /// <summary>
+        ///     Specifies the <see cref="ProcessorOptions"/> to use when processing data sources.
+        /// </summary>
+        /// <param name="options"> <see cref="ProcessorOptions"/> to use for all processors that will process the data sources. </param>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="options"/> is <c>null</c>.
+        /// </exception>
+        public void WithProcessorOptions(ProcessorOptions options)
+        {
+            Guard.NotNull(options, nameof(options));
+
+            this.WithProcessorOptions(new GlobalProcessingOptionsResolver(options));
+        }
+
+        /// <summary>
+        ///     Specifies the <see cref="ProcessorOptions"/> to use when processing data sources.
+        /// </summary>
+        /// <param name="optionsMap">A dictionary which maps <see cref="ProcessorOptions"/> for a given processor which will process the data sources. </param>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="optionsMap"/> is <c>null</c>.
+        /// </exception>
+        public void WithProcessorOptions(IDictionary<Guid, ProcessorOptions> optionsMap)
+        {
+            Guard.NotNull(optionsMap, nameof(optionsMap));
+
+            this.WithProcessorOptions(new ProcessingSourceOptionsResolver(optionsMap));
+        }
+
+        /// <summary>
+        ///     Specifies an <see cref="IProcessingOptionsResolver"/> to provide <see cref="ProcessorOptions"/> to use when processing data sources.
+        /// </summary>
+        /// <remarks>
+        ///     To set processor options for all processing sources use <see cref="WithProcessorOptions(ProcessorOptions)"/>.
+        ///     To set processor options per processing source use <seealso cref="WithProcessorOptions(IDictionary{Guid, ProcessorOptions})"/>.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException">
+        ///     <paramref name="processingOptionsResolver"/> is <c>null</c>.
+        /// </exception>
+        public void WithProcessorOptions(IProcessingOptionsResolver processingOptionsResolver)
```

Comment:
>WithProcessorOptions [](http://example.com/codeflow?start=20&length=20)

Because this is "With" and not "Set", it might make more sense to return "this" to get syntax like this:

var engineCreateInfo = new EngineCreateInfo(...).With(myResolver);

Reply:
Or optionally rename it to a "Set" method.

Reply:
I don't know if this is industry standard, but it's convention in the SDK. If you do a search, you'll see that all "With" methods created/used in this project return something - often time "this".

Reply:
+1 to return `this`. Eventually I would like customers to be able to write

```cs
var engineCreateInfo = new EngineCreateInfo()
    .WithProcessorOptions(foo)
    .WithGroupResolver(bar);
```